app/models/miq_request/purging.rb

```diff
@@ -0,0 +1,32 @@
+class MiqRequest
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        ::Settings.miq_request.history.keep_miq_requests.to_i_with_method.seconds.ago.utc
+      end
+
+      def purge_window_size
+        ::Settings.miq_request.history.purge_window_size
+      end
+
+      def purge_scope(older_than)
+        MiqRequest.where(arel_table[:created_on].lt(older_than))
+      end
+
+      # this is MiqRequest#destroy hook (defined by relationships on MiqRequest)
+      def purge_associated_records(ids)
+        # sorry for the destroy. this needs to be recursive:
+        MiqRequestTask.where(:miq_request_id => ids).destroy_all
+        MiqApproval.where(:miq_request_id => ids).delete_all
+        RequestLog.where(:resource_id => ids).delete_all
+      end
+
+      def purge_method
+        :destroy
```

Comment:
This will ensure the batch of requests are destroyed successfully before moving on. This should help prevent orphans on the other tables.

Reply:
Note, it's worth noting, since we now purge associated records BEFORE the batch records, this will happen first even if there is a dependent => destroy on request_logs.

Reply:
@kbrock @Fryguy see above... thoughts on changing request_logs to `delete_all`...?  purge_associated_records happens first but anything that tries to manually destroy several requests will instantiate a ton of request log objects.  I'm leaning towards `delete_all`.  My two dbs I've seen have 200-800 logs per request.

Reply:
We could put in a delete all. And add a through line with a delete all?

Though, I can't remember how the delete and through works. Remember having issues and removing those. (If you delete the association before the through)

Reply:
Ok, I added delete_all on the association ☝️ https://github.com/ManageIQ/manageiq/pull/21577/files#r2018868626